gobblin-utility/src/main/java/gobblin/util/concurrent/TaskSchedulerFactory.java

```diff
@@ -0,0 +1,54 @@
+package gobblin.util.concurrent;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.Map;
+
+public class TaskSchedulerFactory {
+  private static final Map<String, Class> SCHEDULER_MAP = buildSchedulerMap();
```

Comment:
You are referring to the Guava cache, which can be size bounded, right?  

In this case, how can it leak?